pkg/hintrunner/hinter/scope.go

```diff
@@ -65,36 +86,48 @@ func (sm *ScopeManager) AssignVariables(values map[string]any) error {
 	return nil
 }
 
-func (sm *ScopeManager) GetVariableValue(name string) (any, error) {
+func (sm *ScopeManager) GetScopeValue(name string) (ScopeValue, error) {
 	scope, err := sm.getCurrentScope()
 	if err != nil {
-		return nil, err
+		return ScopeValue{}, err
 	}
 
 	if value, ok := (*scope)[name]; ok {
 		return value, nil
 	}
 
-	return nil, fmt.Errorf("variable %s not found in current scope", name)
+	return ScopeValue{}, fmt.Errorf("variable %s not found in current scope", name)
 }
 
-func (sm *ScopeManager) GetVariableValueAsBigInt(name string) (*big.Int, error) {
-	value, err := sm.GetVariableValue(name)
+func (sm *ScopeManager) GetVariableValueAsFelt(name string) (fp.Element, error) {
+	value, err := sm.GetScopeValue(name)
 	if err != nil {
-		return nil, err
+		return fp.Element{}, err
 	}
 
-	valueBig, ok := value.(*big.Int)
-	if !ok {
-		return nil, fmt.Errorf("value: %s is not a *big.Int", value)
+	return value.Felt, nil
+}
+
+func (sm *ScopeManager) GetVariableValueAsBigInt(name string) (big.Int, error) {
+	value, err := sm.GetScopeValue(name)
+	if err != nil {
+		return big.Int{}, err
 	}
 
-	return valueBig, nil
+	return value.BigInt, nil
 }
 
-func (sm *ScopeManager) getCurrentScope() (*map[string]any, error) {
+func (sm *ScopeManager) getCurrentScope() (*ScopeMap, error) {
 	if len(sm.scopes) == 0 {
 		return nil, fmt.Errorf("expected at least one existing scope")
 	}
 	return &sm.scopes[len(sm.scopes)-1], nil
 }
+
+func SetBigIntScopeValue(value big.Int) ScopeValue {
```

Comment:
I feel like the name can be better. `SetBigIntScopeValue` sounds like its adding a value to scope.

Reply:
Well, it's adding the big.Int/felt value to the ScopeValue

map[string]ScopeValue

![image](https://github.com/NethermindEth/cairo-vm-go/assets/110683318/6cbb28fc-02b8-4e9c-aa22-85f88a74e8eb)